client/src/pages/users.tsx

```diff
@@ -0,0 +1,284 @@
+import LaunchIcon from "@material-ui/icons/Launch";
+import { ColumnDef, ColumnHeader } from "components/column-header";
+import NavBar from "components/nav-bar";
+import { UseUserData, useWithUsers } from "hooks/graphql/use-with-users";
+import withAuthorizationOnly from "hooks/wrap-with-authorization-only";
+import React from "react";
+import { Connection, Edge, User, UserRole } from "types";
+import withLocation from "wrap-with-location";
+import { CLIENT_ENDPOINT } from "api";
+import KeyboardArrowLeftIcon from "@material-ui/icons/KeyboardArrowLeft";
+import KeyboardArrowRightIcon from "@material-ui/icons/KeyboardArrowRight";
+import { ErrorDialog } from "components/dialog";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flexFlow: "column",
+  },
+  container: {
+    flexGrow: 1,
+  },
+  appBar: {
+    height: "10%",
+    top: "auto",
+    bottom: 0,
+  },
+  progress: {
+    position: "fixed",
+    top: "50%",
+    left: "50%",
+  },
+  paging: {
+    marginLeft: "auto",
+    marginRight: "auto",
+  },
+  dropdown: {
+    width: 170,
+  },
+  actionItem: {
+    margin: 10,
+  },
+  normalButton: {
+    "&:hover": {
+      color: theme.palette.primary.main,
+    },
+  },
+}));
+
+const columns: ColumnDef[] = [
+  {
+    id: "name",
+    label: "Name",
+    minWidth: 170,
+    align: "left",
+    sortable: true,
+  },
+  {
+    id: "email",
+    label: "Email",
+    minWidth: 170,
+    align: "left",
+    sortable: true,
+  },
+  {
+    id: "userRole",
+    label: "Role",
+    minWidth: 170,
+    align: "left",
+    sortable: true,
+  },
+  {
+    id: "actions",
+    label: "",
+    minWidth: 0,
+    align: "left",
+    sortable: false,
+  },
+];
+
+function TableFooter(props: {
+  hasNext: boolean;
+  hasPrev: boolean;
+  onNext: () => void;
+  onPrev: () => void;
+}): JSX.Element {
+  const styles = useStyles();
+  const { hasNext, hasPrev, onNext, onPrev } = props;
+  return (
+    <AppBar position="sticky" color="default" className={styles.appBar}>
+      <Toolbar>
+        <div className={styles.paging}>
+          <IconButton data-cy="prev-page" disabled={!hasPrev} onClick={onPrev}>
+            <KeyboardArrowLeftIcon />
+          </IconButton>
+          <IconButton data-cy="next-page" disabled={!hasNext} onClick={onNext}>
+            <KeyboardArrowRightIcon />
+          </IconButton>
+        </div>
+      </Toolbar>
+    </AppBar>
+  );
+}
+
+function UserItem(props: {
+  edge: Edge<User>;
+  i: number;
+  userPagin: UseUserData;
+  userRole: UserRole;
+}): JSX.Element {
+  const { edge, i } = props;
+  const styles = useStyles();
+  const noEditPermission =
+    props.userRole === UserRole.USER ||
+    (edge.node.userRole === UserRole.ADMIN &&
+      props.userRole !== UserRole.ADMIN);
+
+  function handleRoleChange(user: string, permission: string) {
+    props.userPagin.onUpdateUserPermissions(user, permission);
+  }
+
+  function handleLaunchMentor(mentorId: string) {
```

Comment:
ok, just scanning this PR, I've noticed two distinct functions that launch a mentor. Find all such functions and replace them with a single, importable helper function, e.g. `launchMentor(mentorID: string)`